src/server/resolvers.ts

```diff
@@ -0,0 +1,335 @@
+		modifiedAt: new Date().getTime(),
+		shirtSize: args.shirtSize ? toShirtSizeEnum(args.shirtSize) : '',
+	};
+
+	if (user.userType === UserType.Hacker) {
+		const { value } = await models.Hackers.findOneAndUpdate(
+			{ email: user.email },
+			{ $set: newValues },
+			{ returnOriginal: false }
+		);
+		if (!value) throw new UserInputError(`user ${user.email} not found`);
+		return value;
+	} else if (user.userType === UserType.Organizer) {
+		const { value } = await models.Organizers.findOneAndUpdate(
+			{ email: user.email },
+			{ $set: newValues },
+			{ returnOriginal: false }
+		);
+		if (!value) throw new UserInputError(`user ${user.email} not found`);
+		return value;
+	}
+	throw new ApolloError(`updateUser for userType ${user.userType} not implemented`);
+}
+
+async function fetchUser(
+	user: { email: string; userType: string },
+	models: Models
+): Promise<UserDbInterface> {
+	if (user.userType === UserType.Hacker) {
+		const hacker = await models.Hackers.findOne({ email: user.email });
+		if (!hacker) throw new UserInputError(`hacker "${user.email}" not found`);
+		return hacker;
+	} else if (user.userType === UserType.Organizer) {
+		const organizer = await models.Organizers.findOne({ email: user.email });
+		if (!organizer) throw new UserInputError(`user ${user.email} not found`);
+		return organizer;
+	}
+	throw new ApolloError(`updateUser for userType ${user.userType} not implemented`);
+}
+
+export interface Resolvers {
+	ApplicationField: ApplicationFieldResolvers;
+	ApplicationQuestion: ApplicationQuestionResolvers;
+	Hacker: HackerResolvers;
+	Login: LoginResolvers;
+	Mentor: MentorResolvers;
+	Mutation: MutationResolvers;
+	Organizer: OrganizerResolvers;
+	Query: QueryResolvers;
+	Shift: ShiftResolvers;
+	Team: TeamResolvers;
+	User: {
+		__resolveType: (user: UserDbInterface) => 'Hacker' | 'Organizer' | 'Mentor';
+	};
+}
+
+export const resolvers: Resolvers = {
+	ApplicationField: {
+		answer: async field => (await field).answer || null,
+		createdAt: async field => (await field).createdAt.getTime(),
+		id: async field => (await field).id,
+		question: async field => (await field).question,
+	},
+	ApplicationQuestion: {
+		instruction: async question => (await question).instruction || null,
+		note: async question => (await question).note || null,
+		prompt: async question => (await question).prompt,
+	},
+	Hacker: {
+		adult: async hacker => (await hacker).adult || null,
+		createdAt: async hacker => (await hacker).createdAt.getTime(),
+		dietaryRestrictions: async hacker => (await hacker).dietaryRestrictions.map(toDietEnum),
+		email: async hacker => (await hacker).email,
+		firstName: async hacker => (await hacker).firstName,
+		gender: async hacker => (await hacker).gender || null,
+		github: async hacker => (await hacker).github || null,
+		gradYear: async hacker => (await hacker).gradYear || null,
+		id: async hacker => ((await hacker)._id as unknown) as string,
```

Comment:
Can we type this better, something like `bson.ObjectID`?

Reply:
Done.